esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -324,3 +324,251 @@ def __call__(self, cube):
+    """
+    Regrid rectilinear cube onto unstructured mesh.
+
+    Return a new cube with data values calculated using the area weighted
+    mean of data values from rectilinear cube src_cube regridded onto the
+    horizontal mesh of mesh_cube. The dimensions on the cube associated
+    with the grid will replaced by a dimension associated with the mesh.
+    That dimension will be the the first of the grid dimensions, whether
+    it is associated with the x or y coordinate.
+    This function requires that the horizontal dimension of mesh_cube is
+    described by a 2D mesh with data located on the faces of that mesh.
+    This function requires that the horizontal grid of src_cube is
+    rectilinear (i.e. expressed in terms of two orthogonal 1D coordinates).
+    This function also requires that the coordinates describing the
+    horizontal grid have bounds.
+
+    Parameters
+    ----------
+    src_cube : cube
+        A rectilinear instance of iris.cube.Cube that supplies the data,
+        metadata and coordinates.
+    mesh_cube : cube
+        An unstructured instance of iris.cube.Cube that supplies the desired
+        horizontal mesh definition.
+    mdtol : float, optional
+        Tolerance of missing data. The value returned in each element of the
+        returned cube's data array will be masked if the fraction of masked
+        data in the overlapping cells of the source cube exceeds mdtol. This
+        fraction is calculated based on the area of masked cells within each
+        target cell. mdtol=0 means no missing data is tolerated while mdtol=1
+        will mean the resulting element will be masked if and only if all the
+        overlapping cells of the source cube are masked. Defaults to 0.
+
+    Returns
+    -------
+    cube
+        A new iris.cube.Cube instance.
+
+    """
+    regrid_info = _regrid_rectilinear_to_unstructured__prepare(src_cube, mesh_cube)
+    result = _regrid_rectilinear_to_unstructured__perform(src_cube, regrid_info, mdtol)
+    return result
+
+
+class GridToMeshESMFRegridder:
+    """Regridder class for rectilinear to unstructured cubes."""
+
+    def __init__(self, src_mesh_cube, target_grid_cube, mdtol=1):
+        """
+        Create regridder for conversions between source grid and target mesh.
+
+        Parameters
+        ----------
+        src_grid_cube : cube
+            The unstructured iris cube providing the source grid.
+        target_grid_cube : cube
+            The rectilinear iris cube providing the target mesh.
+        mdtol : float, optional
+            Tolerance of missing data. The value returned in each element of
+            the returned array will be masked if the fraction of masked data
+            exceeds mdtol. mdtol=0 means no missing data is tolerated while
+            mdtol=1 will mean the resulting element will be masked if and only
+            if all the contributing elements of data are masked.
+            Defaults to 1.
+
+        """
```

Comment:
I suppose this is checked implicitly by eventually calling `get_xy_dim_coords` which would fail if it were not rectilinear. I suppose there could be a case for providing a clearer error message. I'll look into whether the current error handling is clear enough or if it would benefit from clarification here.